Lib/fontTools/merge.py

```diff
@@ -13,11 +13,33 @@
 import time
 import operator
 import logging
-
+import os
+from fontTools import subset
 
 log = logging.getLogger("fontTools.merge")
 timer = Timer(logger=logging.getLogger(__name__+".timer"), level=logging.INFO)
 
+def _desubtoutinize(otf):
```

Comment:
by importing fontTools.subset.cff module, the CFF table gets a new `desubroutinize` method that you can call here (e.g. we do that in cffsubr: https://github.com/adobe-type-tools/cffsubr/blob/017660be89a7515abaccaa369130560049b232e1/src/cffsubr/__init__.py#L259-L268)
Maybe we should move [that code](https://github.com/fonttools/fonttools/blob/b29ae8003b5ff89b5df935f70ca584ac54112fea/Lib/fontTools/subset/cff.py#L481-L510) to fontTools.cffLib and have it available all the time, not just within the subsetter.

Reply:
Yes, I wasn’t feeling easy about how that code was so indirectly used.

Reply:
Done. I notice that empty subroutines are removed separately using `remove_unused_subroutines()` I wonder if this should be folded into `desubroutinize()` or if there is a reason to keep it separate.

Reply:
> remove_unused_subroutines() I wonder if this should be folded into desubroutinize() or if there is a reason to keep it separate.

well, that's used when desubroutinize is False, so i guess they should stay separate
https://github.com/fonttools/fonttools/blob/9abfc00b71fc2d1aebe5f9f52ce737dac0fe7c05/Lib/fontTools/subset/cff.py#L452-L453

it's also used when --no-hinting